src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1068,6 +1092,44 @@ public IColonyPackageManager getPackageManager()
         return packageManager;
     }
 
+    /**
+     * Get all visiting players.
+     * @return the list.
+     */
+    public List<EntityPlayer> getVisitingPlayers()
```

Comment:
Since you are making a clone anyway: Use an ImmutableList

Reply:
Still not an immutable list. Make it return an ImmutableList so that everybody know that it is an ImmutableList. Also pull it up to IColony please.